### PR TITLE
[fx] Modify solver linearize and add corresponding test

### PR DESCRIPTION
**Pull request by Cypher30**
# What's New

In PR #1496, @super-dainiu suggests that we could use more elegant linearize code and node structure, but at that time we still worried about the time overhead for solver to solve the sequence. After some experiments, I'm sure that this old linearize could be left and use the suggested structure. I also add a test to check if the solver's sequence and annotated result is correct. Lastly, I fix a couple of lines of old inefficient code in `activation_checkpoint_codgen.py` and `ckpt_solver_rotor.py`.

### CODE REVIEW

**Comment by Cypher30** [comment]
Screen Shot 2022-09-01 at 16 35 46
Currently we use MetaTensor implemented by @super-dainiu, which could not be used in torch11, the above is the result of `test_fx`.

File: tests/test_fx/test_ckpt_solvers/test_linearize.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
+from ast import For
+from binhex import REASONABLY_LARGE
+from typing import Callable
```
Comment (super-dainiu) [inline tests/test_fx/test_ckpt_solvers/test_linearize.py:3]: What are these imports?

File: tests/test_fx/test_ckpt_solvers/test_linearize.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
+from ast import For
+from binhex import REASONABLY_LARGE
```
Comment (FrankLeeeee) [inline tests/test_fx/test_ckpt_solvers/test_linearize.py:2]: remove unused imports.

File: tests/test_fx/test_ckpt_solvers/test_linearize.py (hunk 1 above)
Comment (Cypher30) [reply tests/test_fx/test_ckpt_solvers/test_linearize.py:3]: I will clean those imports, probably caused by viscose

File: tests/test_fx/test_ckpt_solvers/test_linearize.py (hunk 2 above)
Comment (Cypher30) [reply tests/test_fx/test_ckpt_solvers/test_linearize.py]: Removed~

**Review by super-dainiu — APPROVED** [review]

**Pull request merged**